### PR TITLE
Changeling Buff

### PR DESCRIPTION
**Pull request by RemieRichards**
## Gameplay changes:
- Changelings now regenerate chemicals while dead, up to half and ONLY HALF of their maximum storage (taking into account their storage upgrades, if applicable)
- Changelings now repair `geneticdamage` while dead, with the lowest value they can be taken to being 50, this number is arbitrary but was based on half the value at which fakedeath stops working for too much `geneticdamage`
- Changeling fakedeath timer is now halved, down from 80 seconds (1 minute 20 seconds) to 40 seconds.
- Changelings now get to see the last 8 messages the people they've absorbed have said (if they didn't say 8, you get how ever many they said):

ling_memory_speech
## Code changes:
- `regenerate()` for the changeling datum now takes a mob argument, so we can check if it's dead or not, so we can apply caps if necessary (It takes a mob argument because the changeling datum jumps around just like the mind datum and there's no one unified method for finding a mob from a changeling besides just adding an arg)
- Life no longer `return .`s because it's redundant and adds a tiny bit of overhead.
- `/living` mobs now record the things they've said in a list, clean text.

### CODE REVIEW

**Comment by Anonus** [comment]
Yesssssss.

**Comment by Cebasiz** [comment]
i don't know. i'd prefer it you'd buff their ability to infiltrate rather than their ability to recover from death. they're already way better at being not-dead than any other type of antagonist, what they need is kit that will help them believably impersonate people, stealthily attack and take the place of important crewmembers, etc.

tangent engage

things that would be helpful:

consolidating some of the stings so they can be applied faster and are more dangerous one on one

giving them the ability to see chatlogs of what their victim was saying recently so they can more reliably impersonate the manner of people

making them able to equip clothes faster

ability to assimilate loyalty implants (maybe)



of course i don't mind buffs whatever shape they take. i'd really like to see changeling become a cool game mode again because i really enjoy playing them. as it is though their only real strategy is combat steroid +armblade killing machine and that isn't really what they are supposed to be.

**Comment by RandomMarine** [comment]
>giving them the ability to see chatlogs of what their victim was saying recently so they can more reliably impersonate the manner of people

Holy shit this.

**Comment by GunHog** [comment]
Or the Personal Hivemind: https://github.com/tgstation/-tg-station/pull/689 it is such a good idea, it really needs a revisit!

**Comment by AlexanderUlanH** [comment]
Personal hivemind is my single favorite rejected feature of all time.

**Comment by RemieRichards** [comment]
Changeling_speech_theft

**Comment by Anonus** [comment]
I think 8 is good. Also you missed a period on the last message there.

**Comment by RemieRichards** [comment]
new_ling_memory_output

Quotation marks suggested by @Ikarrus 
Missing full stop pointed out by @Anonus

**Comment by Lumipharon** [comment]
The recent chat thing is good, and makes sense, but buffing lings while dead, and their regen stasis just makes them more awful to actually put down.

**Comment by freerealestate** [comment]
Does it have to be last 8? Can it just be random messages over a certain amount of characters? I imagine a lot of the most recent messages would be them reacting to you murdering them, and not very long or personal.

**Comment by Iamgoofball** [comment]
idea:

record all their said shit:
but make it accessable via a popup window from a verb to check what that nerd has said

for deeper analysis

it should also log me emotes, because I could totally abuse me to prevent lings from hearing what I said

**Comment by RemieRichards** [comment]
@Iamgoofball discussed in irc but saying here for the benefit of others:
Abusing *Me is considered OOC in IC and has been used for bans previously.

**Comment by RandomMarine** [comment]
Plus it has a little logic issue. A ling would learn that the guy did things like kick a chair in anger. But not that he kicked a downed assistant at some point?

It might be nice in the case of absorbing the mime, but it's already notoriously easy to impersonate them. Would be best to stick with just speech.

**Comment by Madjura** [comment]
@RemieRichards What about *me emotes like: *me shrugs, then clears his throat: "Are you sure about this?" (or similar, basically legit emote and speech combined)

**Comment by Robustin** [comment]
Why does this even have a controversial tag?

The memory recovery is almost universally a good thing and a very nice bit of flavor. The reduced death stuff is just quality of life for the lings. There isn't a single place on the station where you can kill a ling and not have it gibbed/cremated/cuffed/jacketed/lockerwelded within 40 seconds, even the ubiquitous emergency o2 lockers often have an emergency welder inside. If you're worried about mankeys just weld the vent or go to an unvented area until help can come finish the job.

Murdering confirmed lings is not a difficult task 99% of the time, it's more of a ritual celebrating their capture than an actual difficulty facing the crew. I have no problem with a mechanic that makes it more punishing for people who are sloppy with ling executions. 

8 is also fine given 2-3 will probably be screaming for help. This is really just a great change and there's very little reason to limit ling's ability to impersonate victims. It's really at the heart of what ling should be.

**Comment by RemieRichards** [comment]
@Madjura I'm sure that would be classified as IC, but that still wouldn't let you cheat the ling as it takes FAR longer to type out a custom *me+text than just a *me (bannable) or genuine text

@Robustin I added the Controversial tag myself, obviously I want the changes and am pushing for them (The PR, for example) but the controversial tag is a nice way of saying "this changes somethings in a way some people are probably not going to be 100% ok with"
It likely won't affect the merging of the PR unless hundreds of people come out of the woodwork to confirm it's controversial nature. 

tl;dr it's mainly for categorisation and premature covering of my ass.
